app/src/main/java/com/rkm/tasky/ui/component/TextField.kt

```diff
@@ -0,0 +1,176 @@
+                    Text(
+                        text = stringResource(R.string.text_field_email_hint),
+                        style = TextStyle(
+                            color = TextFieldHintColor,
+                            fontSize = MaterialTheme.typography.bodyLarge.fontSize,
+                        )
+                    )
+                }
+                innerTextField()
+            }
+        )
+
+        if(isValidEmail) {
+            Icon(
+                imageVector = Icons.Rounded.Check,
+                modifier = Modifier
+                    .weight(1f)
+                    .padding(16.dp),
+                contentDescription = null,
+                tint = TextFieldCheckColor
+            )
+        }
+    }
+
+}
+
+@Composable
+fun PasswordTextField(
+    password: String,
+    onPasswordChange: (String) -> Unit,
+    showPassword: Boolean,
+    modifier: Modifier
+) {
+
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .height(75.dp)
+            .background(color = LightGreyTextFieldBackground, shape = RoundedCornerShape(8.dp)),
+        verticalAlignment = Alignment.CenterVertically
+    )
+    {
+        BasicTextField(
+            value = password,
+            onValueChange = onPasswordChange,
+            modifier = Modifier
+                .weight(5f)
+                .padding(16.dp),
+            textStyle = TextStyle(
+                color = TextFieldFontColor,
+                fontSize = MaterialTheme.typography.bodyLarge.fontSize,
+            ),
+            visualTransformation = if (showPassword) VisualTransformation.None else PasswordVisualTransformation(),
+            decorationBox = { innerTextField ->
+                if (password.isEmpty()) {
+                    Text(
+                        text = stringResource(R.string.text_field_password_hint),
+                        style = TextStyle(
+                            color = TextFieldHintColor,
+                            fontSize = MaterialTheme.typography.bodyLarge.fontSize,
+                        )
+                    )
+                }
+                innerTextField()
+            }
+        )
+
+        if(showPassword) {
+            Icon(
+                painter = painterResource(id = R.drawable.round_visibility_24),
+                modifier = Modifier
+                    .weight(1f)
+                    .padding(16.dp),
+                contentDescription = null,
+                tint = TextFieldHintColor
+            )
+        } else {
+            Icon(
+                painter = painterResource(id = R.drawable.round_visibility_off_24),
+                modifier = Modifier
+                    .weight(1f)
+                    .padding(16.dp),
+                contentDescription = null,
+                tint = TextFieldHintColor
+            )
+        }
+    }
+
+}
+
+@Preview(
+    showBackground = false,
+)
+@Composable
+fun EmailValidationTextFieldPreview() {
```

Comment:
Nit, marking previews as private will prevent it being seen or used outside of this file